include/signature_subpacket/embedded.h

```diff
@@ -66,14 +66,14 @@ namespace pgp::signature_subpacket {
              *
              *  @param  other   The object to compare with
              */
-            bool operator==(const embedded &other) const noexcept;
+            bool operator==(const embedded &other) const;
```

Comment:
Not a fan of this change. I assume you did this because `contained()` may throw if the object was moved-away from. I'd rather change it so `contained()` has the precondition it may not be a moved-from instance (and is then marked noexcept).

Reply:
That seems disingenuous, since the point of `noexcept` is that it may never throw, regardless of preconditions. We may change the check instead so that it can compare objects that have been moved from.

This would preserve the `noexcept` for the comparison operators but would actually uphold the contract.

Reply:
@liptakmatyas @vicsn opinions? We appear to fundamentally disagree on what `noexcept` should mean: either "never throw at all and always be safe" or "never throw unless the object is in an undefined state".

Reply:
I'll leave the noexcept semantics to omartijn.

Reply:
@bertptrs That's not exactly what I said. What I said is that calling the function on a moved-from object results in undefined behaviour. That doesn't necessarily mean it throws.